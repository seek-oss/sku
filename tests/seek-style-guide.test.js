/* eslint-disable jest/expect-expect */
const path = require('path');
const dirContentsToObject = require('../test/utils/dirContentsToObject');
const runSkuScriptInDir = require('../test/utils/runSkuScriptInDir');
const waitForUrls = require('../test/utils/waitForUrls');

const appDir = path.dirname(
  require.resolve('@fixtures/seek-style-guide/sku.config.js'),
);
const distDir = path.resolve(appDir, 'dist');

const assertStorybookContent = async (url) => {
  const page = await browser.newPage();
  await page.goto(url, { waitUntil: 'networkidle2' });

  const content = await page.evaluate(() => {
    const svg = window.document
      .querySelector('iframe')
      .contentDocument.querySelector('[data-automation-svg] svg');

    const textElement = window.document
      .querySelector('iframe')
      .contentDocument.querySelector('[data-automation-text]');
    const text = textElement.innerText;

    return { svg, text };
  });

  expect(content.svg).not.toEqual(null);
  expect(content.text).toEqual('Hello world!');
};

describe('seek-style-guide', () => {
  beforeAll(async () => {
    await runSkuScriptInDir('build', appDir);
  });

  it('should generate the expected files', async () => {
    const files = await dirContentsToObject(distDir);
    expect(files).toMatchSnapshot();
  });

  it('should handle seek-style-guide in tests', async () => {
    const { childProcess } = await runSkuScriptInDir('test', appDir);
    expect(childProcess.exitCode).toEqual(0);
  });

  describe('storybook', () => {
    const storybookUrl = 'http://localhost:8083';
    let server;

    beforeAll(async () => {
      server = await runSkuScriptInDir('storybook', appDir, ['--ci']);
      await waitForUrls(storybookUrl);
    });

    afterAll(async () => {
      await server.kill();
    });

    it('should start a storybook server', async () => {
      await assertStorybookContent(storybookUrl);
      expect.assertions(2);
    });
  });
});
