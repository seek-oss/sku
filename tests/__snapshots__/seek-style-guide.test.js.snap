// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`seek-style-guide should generate the expected files 1`] = `
{
  "747-5cbea71031dd0a29e06b.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "externals.json": "[
  "crypto",
  "fs",
  "os",
  "path",
  "stream",
  "tty",
  "url",
  "util"
]",
  "index.html": SCRIPTS: [
    "/runtime-809e1ad97457d5b9bab0.js",
    "/747-5cbea71031dd0a29e06b.js",
    "/main-5cdeca43d96c2412df5c.js",
  ]
CSS: [
    "/main-9f810ad2db5899b4e432.css",
  ]
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      My Awesome Project
    </title>
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1"
    >
    <link
      data-chunk="main"
      rel="stylesheet"
      href="styles[0]"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[0]"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[1]"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[2]"
    >
  </head>
  <body>
    <div id="app">
      <div class="LA4NIPb">
        <div class="kKyygCZ">
          <h1>
            SEEK - Australia&#x27;s no. 1 jobs, employment, career and recruitment site
          </h1>
        </div>
        <div class>
          <div class="QqjGnE2">
            <span class="a3YqhIM ucedCT7 UrE2lfH">
              <span class>
                <span data-automation-svg="true">
                  <span class="EkNtYHl">
                    <svg
                      class="n5z9w0i"
                      xmlns="http://www.w3.org/2000/svg"
                      width="16"
                      height="16"
                      viewbox="0 0 1024 900"
                      focusable="false"
                    >
                      <path d="M718 188q11 0 22 1.5t22 4.5q16 4 30.5 12t28.5 19q15 13 27 28t21 34h1l3 7q13 27 20 59.5t7 68.5v3q0 15-4.5 32.5T881 496q-11 24-26 47.5T821 590q-39 48-85.5 94T636 774q-46 38-85 67.5T480 891q-31-20-70-49.5T325 774h-1q-52-43-98.5-89T139 590q-19-23-34-46.5T79 496q-10-21-14.5-38.5T60 425v-3q0-37 7-69.5T87 293h1q0-2 1-3.5t2-3.5q10-18 21.5-33t26.5-28q14-11 28.5-19t30.5-12q11-3 22.5-4.5T242 188q24 0 45.5 7t43.5 21q15 9 28 21t24 26q12 15 22.5 32.5T425 333l56 129 54-129q9-20 19-37t22-32q11-14 24.5-26t28.5-22h1q21-14 43-21t45-7zm0-60q-32 0-62 9.5T597 165q-38 25-67.5 61T480 309q-21-47-50-83t-66-61q-29-18-59.5-27.5T242 128q-14 0-29 2t-30 6q-22 6-43 16.5T100 179t-34.5 36T38 259q-1 1-1.5 2t-.5 2l-2 2v1q-17 34-25.5 73T0 422v2q0 22 6 46t19 51q12 27 29 54t39 53q42 52 90.5 100T286 820q49 41 90 71.5t74 51.5q14 9 21.5 13.5t8.5 4.5q1 0 8.5-4.5T510 944q33-22 74-53t90-70q55-46 103.5-94t89.5-99q22-26 39-52.5t29-53.5q13-27 19-51.5t6-46.5v-2q-1-44-9-83t-24-72l-4-8q-13-24-28.5-43.5T860 180t-39.5-27-43.5-17q-15-4-30-6t-29-2z">
                      </path>
                    </svg>
                  </span>
                </span>
                <span data-automation-text="true">
                  Hello world!
                </span>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS__"
      type="application/json"
    >
      []
    </script>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS___ext"
      type="application/json"
    >
      {"namedChunks":[]}
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[0]"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[1]"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[2]"
    >
    </script>
  </body>
</html>,
  "main-5cdeca43d96c2412df5c.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "main-9f810ad2db5899b4e432.css": a,
abbr,
acronym,
address,
applet,
article,
aside,
audio,
b,
big,
blockquote,
body,
canvas,
caption,
center,
cite,
code,
dd,
del,
details,
dfn,
div,
dl,
dt,
em,
embed,
fieldset,
figcaption,
figure,
footer,
form,
h1,
h2,
h3,
h4,
h5,
h6,
header,
hgroup,
html,
i,
iframe,
img,
ins,
kbd,
label,
legend,
li,
mark,
menu,
nav,
object,
ol,
output,
p,
pre,
q,
ruby,
s,
samp,
section,
small,
span,
strike,
strong,
sub,
summary,
sup,
table,
tbody,
td,
tfoot,
th,
thead,
time,
tr,
tt,
u,
ul,
var,
video {
  border: 0;
  font-size: 100%;
  font: inherit;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}
article,
aside,
details,
figcaption,
figure,
footer,
header,
hgroup,
menu,
nav,
section {
  display: block;
}
body {
  line-height: 1;
}
ol,
ul {
  list-style: none;
}
blockquote,
q {
  quotes: none;
}
blockquote:after,
blockquote:before,
q:after,
q:before {
  content: "";
  content: none;
}
table {
  border-collapse: collapse;
  border-spacing: 0;
}
mark {
  background-color: transparent;
  color: inherit;
}
html {
  min-height: 100%;
}
.LA4NIPb,
html {
  background-color: #eee;
}
.LA4NIPb {
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  box-sizing: border-box;
  color: #1c1c1c;
  font-family: Roboto, Helvetica Neue, HelveticaNeue, Helvetica, Arial,
    sans-serif;
  font-size: 10px;
}
.LA4NIPb *,
.LA4NIPb :after,
.LA4NIPb :before {
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  -webkit-tap-highlight-color: transparent;
  box-sizing: inherit;
  font-family: inherit;
}
.LA4NIPb input[type="search"]::-webkit-search-decoration {
  -webkit-appearance: none;
}
.nl7aQqy {
  height: 100%;
}
.kKyygCZ {
  clip: rect(1px, 1px, 1px, 1px);
  pointer-events: none;
  position: absolute !important;
}
.QqjGnE2 {
  margin: 0 auto;
  max-width: 940px;
}
.a3YqhIM {
  display: block;
  padding-bottom: 12px;
}
.a3YqhIM .a3YqhIM {
  outline: 2px solid red;
  position: relative;
}
.a3YqhIM .a3YqhIM:after {
  background: red;
  color: #fff;
  content: "ERROR: Multiple <Text> components nested";
  font-weight: 700;
  padding: 0 20px;
  position: absolute;
  right: 0;
  top: 0;
}
.ucedCT7 {
  padding-bottom: 6px;
}
@media only screen and (max-width: 739px) {
  .ucedCT7:not(.UrE2lfH) {
    font-size: 16px;
    font-weight: 400;
    line-height: 24px;
  }
}
@media only screen and (min-width: 740px) {
  .ucedCT7:not(.UrE2lfH) {
    font-size: 16px;
    font-weight: 400;
    line-height: 24px;
  }
}
@media only screen and (max-width: 739px) {
  .ucedCT7.UrE2lfH {
    font-size: 16px;
    font-weight: 400;
    line-height: 24px;
    transform: translateY(0.41em);
  }
}
@media only screen and (min-width: 740px) {
  .ucedCT7.UrE2lfH {
    font-size: 16px;
    font-weight: 400;
    line-height: 24px;
    transform: translateY(0.41em);
  }
}
.YFwEAvh {
  padding-bottom: 6px;
}
@media only screen and (max-width: 739px) {
  .YFwEAvh:not(.UrE2lfH) {
    font-size: 14px;
    font-weight: 400;
    line-height: 18px;
  }
}
@media only screen and (min-width: 740px) {
  .YFwEAvh:not(.UrE2lfH) {
    font-size: 14px;
    font-weight: 400;
    line-height: 18px;
  }
}
@media only screen and (max-width: 739px) {
  .YFwEAvh.UrE2lfH {
    font-size: 14px;
    font-weight: 400;
    line-height: 18px;
    transform: translateY(0.30285714em);
  }
}
@media only screen and (min-width: 740px) {
  .YFwEAvh.UrE2lfH {
    font-size: 14px;
    font-weight: 400;
    line-height: 18px;
    transform: translateY(0.30285714em);
  }
}
.sAUCobp {
  padding-bottom: 6px;
}
@media only screen and (max-width: 739px) {
  .sAUCobp:not(.UrE2lfH) {
    font-size: 18px;
    font-weight: 400;
    line-height: 30px;
  }
}
@media only screen and (min-width: 740px) {
  .sAUCobp:not(.UrE2lfH) {
    font-size: 18px;
    font-weight: 400;
    line-height: 30px;
  }
}
@media only screen and (max-width: 739px) {
  .sAUCobp.UrE2lfH {
    font-size: 18px;
    font-weight: 400;
    line-height: 30px;
    transform: translateY(0.49333333em);
  }
}
@media only screen and (min-width: 740px) {
  .sAUCobp.UrE2lfH {
    font-size: 18px;
    font-weight: 400;
    line-height: 30px;
    transform: translateY(0.49333333em);
  }
}
@media only screen and (max-width: 739px) {
  .q8WiTVW:not(.UrE2lfH) {
    font-size: 18px;
    font-weight: 700;
    line-height: 24px;
  }
}
@media only screen and (min-width: 740px) {
  .q8WiTVW:not(.UrE2lfH) {
    font-size: 18px;
    font-weight: 700;
    line-height: 24px;
  }
}
@media only screen and (max-width: 739px) {
  .q8WiTVW.UrE2lfH {
    font-size: 18px;
    font-weight: 700;
    line-height: 24px;
    transform: translateY(0.32666667em);
  }
}
@media only screen and (min-width: 740px) {
  .q8WiTVW.UrE2lfH {
    font-size: 18px;
    font-weight: 700;
    line-height: 24px;
    transform: translateY(0.32666667em);
  }
}
@media only screen and (max-width: 739px) {
  .DBCZQ2H:not(.UrE2lfH) {
    font-size: 21px;
    font-weight: 700;
    line-height: 30px;
  }
}
@media only screen and (min-width: 740px) {
  .DBCZQ2H:not(.UrE2lfH) {
    font-size: 28px;
    font-weight: 700;
    line-height: 36px;
  }
}
@media only screen and (max-width: 739px) {
  .DBCZQ2H.UrE2lfH {
    font-size: 21px;
    font-weight: 700;
    line-height: 30px;
    transform: translateY(0.37428571em);
  }
}
@media only screen and (min-width: 740px) {
  .DBCZQ2H.UrE2lfH {
    font-size: 28px;
    font-weight: 700;
    line-height: 36px;
    transform: translateY(0.30285714em);
  }
}
@media only screen and (max-width: 739px) {
  .NDv4mwF:not(.UrE2lfH) {
    font-size: 21px;
    font-weight: 700;
    line-height: 30px;
  }
}
@media only screen and (min-width: 740px) {
  .NDv4mwF:not(.UrE2lfH) {
    font-size: 22px;
    font-weight: 700;
    line-height: 30px;
  }
}
@media only screen and (max-width: 739px) {
  .NDv4mwF.UrE2lfH {
    font-size: 21px;
    font-weight: 700;
    line-height: 30px;
    transform: translateY(0.37428571em);
  }
}
@media only screen and (min-width: 740px) {
  .NDv4mwF.UrE2lfH {
    font-size: 22px;
    font-weight: 700;
    line-height: 30px;
    transform: translateY(0.34181818em);
  }
}
@media only screen and (max-width: 739px) {
  .yGG7Sh5:not(.UrE2lfH) {
    font-size: 21px;
    font-weight: 700;
    line-height: 30px;
  }
}
@media only screen and (min-width: 740px) {
  .yGG7Sh5:not(.UrE2lfH) {
    font-size: 42px;
    font-weight: 400;
    line-height: 48px;
  }
}
@media only screen and (max-width: 739px) {
  .yGG7Sh5.UrE2lfH {
    font-size: 21px;
    font-weight: 700;
    line-height: 30px;
    transform: translateY(0.37428571em);
  }
}
@media only screen and (min-width: 740px) {
  .yGG7Sh5.UrE2lfH {
    font-size: 42px;
    font-weight: 400;
    line-height: 48px;
    transform: translateY(0.23142857em);
  }
}
.JfsMHgO {
  font-weight: 300;
}
.BO2SZNS {
  padding-bottom: 0;
}
.wP0g3sg {
  display: list-item;
  list-style-type: disc;
  margin-left: 30px;
}
.yojLet0 {
  color: #169400;
}
.igeymRT {
  color: #e60278;
}
.kxauBAq {
  color: #747474;
}
.XggRrIV {
  background-color: rgba(30, 144, 255, 0.12);
  border-radius: 2px;
  display: inline-block;
  margin: -2px;
  padding: 2px;
}
.pxuB37w {
  color: #404040;
}
.MOwZ_VU {
  background-color: rgba(230, 2, 120, 0.12);
}
.FGcPm1q {
  color: #9556b7;
}
.pXhk0hH {
  font-weight: 700;
}
.eDwxxJg {
  font-weight: 400;
}
.C4hDKnh {
  animation-duration: 0.4s;
  animation-name: jdTC3YO;
}
@keyframes jdTC3YO {
  0% {
    transform: scale(0.5);
  }
  65% {
    transform: scale(1.5);
  }
  85% {
    transform: scale(0.8);
  }
  to {
    transform: scale(1);
  }
}
.EkNtYHl {
  display: inline-block;
}
.n5z9w0i {
  fill: currentColor;
  height: 16px;
  width: 16px;
}
.KlXqWjP {
  height: 29.4px;
  width: 29.4px;
}
@media only screen and (max-width: 739px) {
  .KlXqWjP {
    height: 14.7px;
    width: 14.7px;
  }
}
.j7b5G9y {
  height: 19.6px;
  width: 19.6px;
}
@media only screen and (max-width: 739px) {
  .j7b5G9y {
    height: 14.7px;
    width: 14.7px;
  }
}
.hJigY6O {
  height: 12.6px;
  width: 12.6px;
}
@media only screen and (max-width: 739px) {
  .hJigY6O {
    height: 12.6px;
    width: 12.6px;
  }
}
.EgQnENm {
  height: 15.4px;
  width: 15.4px;
}
@media only screen and (max-width: 739px) {
  .EgQnENm {
    height: 14.7px;
    width: 14.7px;
  }
}
.O5TUWct {
  height: 12.6px;
  width: 12.6px;
}
@media only screen and (max-width: 739px) {
  .O5TUWct {
    height: 12.6px;
    width: 12.6px;
  }
}
.fP1g34I {
  height: 11.2px;
  width: 11.2px;
}
@media only screen and (max-width: 739px) {
  .fP1g34I {
    height: 11.2px;
    width: 11.2px;
  }
}
.XXOkB_c {
  height: 9.8px;
  width: 9.8px;
}
@media only screen and (max-width: 739px) {
  .XXOkB_c {
    height: 9.8px;
    width: 9.8px;
  }
}
.Ym8IMVv {
  height: 12.6px;
  width: 12.6px;
}
@media only screen and (max-width: 739px) {
  .Ym8IMVv {
    height: 12.6px;
    width: 12.6px;
  }
}
,
  "runtime-809e1ad97457d5b9bab0.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
}
`;
